controllers/storagecluster/storageclient.go

```diff
@@ -126,3 +142,19 @@ func (s *storageClient) updateClientConfigMap(r *StorageClusterReconciler, names
 
 	return nil
 }
+
+// getCephNetworkAnnotationValue returns the network annotation value for the given StorageCluster NetworkSpec.
+func getCephNetworkAnnotationValue(cephNetworkSpec *rookCephv1.NetworkSpec, scNamespace string) (string, error) {
+	if len(cephNetworkSpec.Selectors) != 0 {
+		networkSelectionElement, err := cephNetworkSpec.GetNetworkSelection(scNamespace, rookCephv1.CephNetworkType(cephPublicNetworkType))
+		if err != nil || networkSelectionElement == nil {
+			return "", fmt.Errorf("failed to get network selection element: %v", err)
+		}
+		nwAnnotation, err := json.Marshal([]*nadv1.NetworkSelectionElement{networkSelectionElement})
+		if err != nil {
+			return "", fmt.Errorf("failed to marshal network selection elements: %v", err)
+		}
+		return string(nwAnnotation), nil
+	}
+	return "", fmt.Errorf("invalid ceph network spec")
```

Comment:
Can you please flip the condition so the last return will be the "happy" return?